sdk/src/wallet/core/builder.rs

```diff
@@ -166,13 +166,17 @@ where
             true
         };
 
+        // Panic: client options must exist at this point
+        let client_options = self.client_options.as_ref().unwrap();
+
         // May use a previously stored secret manager if it wasn't provided
         if self.secret_manager.is_none() {
-            let secret_manager = loaded_wallet_builder
+            self.secret_manager = loaded_wallet_builder
                 .as_ref()
                 .and_then(|builder| builder.secret_manager.clone());
-
-            self.secret_manager = secret_manager;
+            if self.secret_manager.is_none() {
+                return Err(crate::wallet::Error::MissingParameter("secret_manager"));
+            }
         }
```

Comment:
```suggestion
        if self.secret_manager.is_none() {
            self.secret_manager.replace(
                loaded_wallet_builder
                    .as_ref()
                    .ok_or(crate::wallet::Error::MissingParameter("secret_manager"))?
                    .secret_manager
                    .clone()
            );
        }
```

Reply:
Sorry :trollface: "Applying suggestions on deleted lines is currently not supported. Please  and try again."

I'll add the change locally.

Reply:
Done